File: benchmark/c/cpw/primality_test_set_5using_lucas_lehmer_series/primality_test_set_5using_lucas_lehmer_series.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int primality_test_set_5using_lucas_lehmer_series ( int p ) ;
int primality_test_set_5using_lucas_lehmer_series ( int p ) {
  long long checkNumber = pow ( 2, p ) - 1;
  long long nextval = 4 % checkNumber;
  for ( int i = 1;
  i < p - 1;
  i ++ ) nextval = ( nextval * nextval - 2 ) % checkNumber;
  if(nextval == 0) return 1;
  else return 0;
}

File: translation_result/else_bms/fail/c/cpw/primality_test_set_5using_lucas_lehmer_series/gpt4/Hinted/primality_test_set_5using_lucas_lehmer_series.rs


fn primality_test_set_5using_lucas_lehmer_series(p: i32) -> i32 {
    let check_number: i64 = (2i64).pow(p as u32) - 1;
    let mut next_val: i64 = 4 % check_number;

    for _ in 1..(p - 1) {
        next_val = (next_val.wrapping_mul(next_val) - 2) % check_number;
    }

    if next_val == 0 {
        1
    } else {
        0
    }
}
